repo_tag.go

```diff
@@ -195,6 +195,12 @@ func (r *Repository) Tags(opts ...TagsOptions) ([]string, error) {
 //
 // Docs: https://git-scm.com/docs/git-tag
 type CreateTagOptions struct {
+	// Annotated marks a tag as annotated rather than lightweight.
+	Annotated bool
+	// Message specifies a tagging message for the annotated tag. It is ignored when tag is not annotated.
+	Message string
```

Comment:
I see, however this would make the message mandatory, or we would need some other way to create annotated tag with empty message.

I actually converted the `-a` option available in git-tag command to a boolean, because we specify the tag name in the function signature, which is either set to the value of the `-a` option or as an argument for lightweight tags.

Reply:
Oh, did not realize message can be empty, then SGTM!

Reply:
When `CreateTagOptions{Annotated: true}`, the git command result:
`GIT_COMMITTER_NAME=$author.name GIT_COMMITTER_EMAIL=$author.email git tag -a $name --message $message $rev`

The `Message` is optional, so the annotated tag will hold an empty or non-empty message.
The `Author` is mandatory, which is enforced by `git` itself, so did not prepend any validation in `CreateTag`.
Similarly, tag `name` is mandatory and this is enforced by `git`.

when `CreateTagOptions{Annotated: false}` (default), git command will be:
`git tag $name $rev`

I pushed an improvement in this PR (95805513939c8f1d450884188ef85b07a5ddf367) to exactly match the above explanation.